internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,340 @@
+
+	dnsResolverBuilder := resolver.Get("dns")
+	resolver.Register(mr)
+
+	t.Cleanup(func() { resolver.Register(dnsResolverBuilder) })
+	return mr
+}
+
+// proxyAddressWithTargetAttribute creates a resolver.Address for the proxy,
+// adding the target address as an attribute.
+func proxyAddressWithTargetAttribute(proxyAddr string, targetAddr string) resolver.Address {
+	addr := resolver.Address{Addr: proxyAddr}
+	addr = proxyattributes.Populate(addr, proxyattributes.Options{
+		User:        nil,
+		ConnectAddr: targetAddr,
+	})
+	return addr
+}
+
+// Tests the scenario where proxy is configured and the target URI contains the
+// "dns" scheme and target resolution is enabled. The test verifies that the
+// addresses returned by the delegating resolver combines the addresses
+// returned by the proxy resolver and the target resolver.
+func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T) {
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
+			return &url.URL{
+				Scheme: "https",
+				Host:   envProxyAddr,
+			}, nil
+		}
+		return nil, nil
+	}
+	originalhpfe := internal.HTTPSProxyFromEnvironmentForTesting
+	internal.HTTPSProxyFromEnvironmentForTesting = hpfe
+	defer func() {
+		internal.HTTPSProxyFromEnvironmentForTesting = originalhpfe
+	}()
+
+	mrTarget := setupDNS(t) // Manual resolver to control the target resolution.
+	target := mrTarget.Scheme() + ":///" + targetTestAddr
+	mrProxy := setupDNS(t) // Set up a manual DNS resolver to control the proxy address resolution.
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	if _, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mrTarget, true); err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	mrProxy.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedProxyTestAddr1},
+			{Addr: resolvedProxyTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	select {
+	case <-stateCh:
+		t.Fatalf("Delegating resolver invoked UpdateState before both the proxy and target resolvers had updated their states.")
+	case <-time.After(defaultTestShortTimeout):
+	}
+
+	mrTarget.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedTargetTestAddr1},
+			{Addr: resolvedTargetTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the expected address.
+	wantState := resolver.State{Addresses: []resolver.Address{
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr1),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr2),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr1),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr2),
+	},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
```

Comment:
Write this block as follows:

```golang
	wantState := resolver.State{
		Addresses: []resolver.Address{
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr1),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr2),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr1),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr2),
		},
		ServiceConfig: &serviceconfig.ParseResult{},
	}
```

This ensures that matching braces are correctly aligned.

Reply:
DOne